File: Easy/206. Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        std::stack<ListNode*> nodes;
		
		ListNode *run = head;
		while(run){
			nodes.push(run);
			run = run->next;
		}
		
		ListNode *preHead = new ListNode(0);
		run = preHead;
		while(!nodes.empty()){
			run->next = nodes.top();
            nodes.pop();
			run = run->next;
		}
		run->next = NULL;
		
		return preHead->next;
    }
};

File: Easy/206. Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        nodes = []
        
        run = head
        while run:
            nodes.append(run)
            run = run.next
        
        preHead = ListNode(0)
        run = preHead
        while nodes:
            run.next = nodes.pop()
            run = run.next
        run.next = None
            
        return preHead.next
        
